pkg/registry/core/service/rest_test.go

```diff
@@ -432,10 +455,19 @@ func TestServiceRegistryExternalService(t *testing.T) {
 	}
 	srv, err := registry.GetService(ctx, svc.Name, &metav1.GetOptions{})
 	if err != nil {
-		t.Errorf("Unexpected error: %v", err)
+		t.Fatalf("Unexpected error: %v", err)
```

Comment:
a fatal here means we won't run the cleanup at the end... is that intentional? will that mess up other tests?